framework/platform/bpl/uci/db/bpl_db_uci.cpp

```diff
@@ -3,10 +3,40 @@
 #include <mapf/common/logger.h>
 #include <mapf/common/utils.h>
 
+extern "C" {
+#include <uci.h>
+}
+
 namespace beerocks {
 namespace bpl {
 namespace db {
 
+constexpr ssize_t MAX_UCI_BUF_LEN  = 64;
+constexpr char package_path[]      = "%s";
+constexpr char section_path[]      = "%s.%s";
+constexpr char section_type_path[] = "%s.%s=%s";
+constexpr char option_path[]       = "%s.%s.%s";
+constexpr char option_value_path[] = "%s.%s.%s=%s";
+
+std::string uci_get_error(struct uci_context *ctx)
+{
+    char *err_buf;
+    uci_get_errorstr(ctx, &err_buf, "");
+    return std::string(err_buf);
```

Comment:
in case `err_buf` is `nullptr` you will cause an exception.   
you can see proof here:   https://repl.it/repls/ShabbyBurlyWorkers#main.cpp   
check for null

Reply:
Added a check, and in case `uci_get_errorstr` does not set `err_buf` it'll return an empty string.